fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/synonym/SynonymAnalyzer.java

```diff
@@ -121,7 +113,11 @@ public Analyzer getIndexAnalyzer(@Nonnull Index index) {
         @Nonnull
         @Override
         public Analyzer getQueryAnalyzer(@Nonnull Index index, @Nonnull Analyzer indexAnalyzer) {
-            return new SynonymAnalyzer(StandardAnalyzer.STOP_WORDS_SET);
+            String name = index.getOption(IndexOptions.TEXT_SYNONYM_SET_NAME_OPTION);
+            if (name == null) {
+                name = USEnglishSynonymMapConfig.CONFIG_NAME;
+            }
+            return new SynonymAnalyzer(EnglishAnalyzer.ENGLISH_STOP_WORDS_SET, name);
```

Comment:
This seems weird & problematic. Perhaps better as a followup, but it seems like synonyms and stop words should be coupled, and registered together. Or at least, both being specified by index options. For example, this would not be the right behavior, if you are indexing non-english, you might get English stop words but, e.g. French synonyms.

Reply:
I agree, we'll need to add the stop words to the config once we start supporting other languages.